hypothesis-python/tests/nocover/test_conjecture_engine.py

```diff
@@ -302,4 +303,4 @@ def shrinker(data):
         (1, 2),
         (1, 3),
         (2, 3),
-    ]
+    }.issubset(blocks)
```

Comment:
Changing this to a subset check mostly undermines the intention behind this particular test, which is that you can look at the test vector and say “yes, that's exactly what the generator should be producing, in order”.

(Changing the other tests is fine, because the order doesn't really matter in those.)

I think the correct approach here is to:

- Go back to the original array-based check.
- Change the “shrink” condition from `(0, 6)` to something like `(3, 4)`, since that occurs much earlier in the new iteration order.
- Update the test vector to `[(0, 1), (1, 2), (2, 3), (3, 4), (0, 2), (1, 3), (0, 3)]`.

This simulates a shrinker pass where we iterate up to `(3, 4)` (in the new order), find a valid shrink, and then keep iterating over the new block structure without anything weird happening.

Reply:
Hmm. Your point is well taken, but I'm worried about how brittle that is to further changes in the order. e.g. it would maybe be reasonable to change this to run in a random order in future. Let me see if I can come up with something better though.

Reply:
How do you feel about the new test? It's agnostic about the order but I think it's a better test than the original in terms of actually testing the essential properties of the iteration.